### ISSUE TITLE
Font sizes of headings cannot be overridden in `MenuSelectHeading`

### ISSUE THREAD

**Issue by RuckieOfficial** [sjdemartini/mui-tiptap#186]
### Which packages did you experience the bug in?

starter-kit

### What Tiptap version are you using?

2.1.13

### What’s the bug you are facing?

I have default settings and in the heading menu select heading h2 is 0.75 rem instead of 1.75rem.

![image](https://github.com/ueberdosis/tiptap/assets/43410493/6c0d9c88-62df-4475-9fb9-6fb9404f3e99)

`extensions={[StarterKit, ...extension]}`

In my custom extensions is not anything, that overrides heading config.

### What browser are you using?

Chrome

### What did you expect to happen?

I expected that font sizes go chronologically.

### Did you update your dependencies?

- [x] Yes, I’ve updated my dependencies to use the latest version of all packages.

**Comment by RuckieOfficial** [comment]
Weird, seems like it takes mui h4 typography settings.

![image](https://github.com/sjdemartini/mui-tiptap/assets/43410493/155f2ccf-0209-45e1-a5f5-e39ef967509d)

![image](https://github.com/sjdemartini/mui-tiptap/assets/43410493/9daf1fb3-b272-4dbd-9fb2-548a72f5acd5)

**Comment by RuckieOfficial** [comment]
Update1: Right I see it uses mui theme heading sizes but skipped.

`"& h2": {      ...(0, import_omit.default)(theme.typography.h4, ["lineHeight"]),
      fontWeight: 500
    },
    "& h3": {
      ...(0, import_omit.default)(theme.typography.h5, ["lineHeight"]),
      fontWeight: 500
    },
    "& h4": {
      ...(0, import_omit.default)(theme.typography.h6, ["lineHeight"]),
      fontWeight: 500
    },`

**Comment by sjdemartini** [comment]
@RuckieOfficial I'm not sure I entirely follow what the issue you're reporting is. Is it that you want each heading style to match the MUI default heading font sizes? For what it's worth, I left a comment in the code about my decision on those styles:

https://github.com/sjdemartini/mui-tiptap/blob/9e4bbb5ab66294ce654ba2e7c141d21fe1e72325/src/styles.ts#L51-L53

In other words, I found the MUI default theme styles for h1 to be useful for something like a landing page hero, but not practical inside a document or similar (like a Tiptap rich text editor), hence choosing smaller heading font sizes all the way down. There's quite a lot to cover in the README as is, so I haven't documented stuff that seems more minor, like decisions around particular default styles in mui-tiptap.

You can override styles for anything in the rich text editor, like in this example (same as in the CodeSandbox linked from the README):

https://github.com/sjdemartini/mui-tiptap/blob/0ad9992eaa106acb0526dd9766aa45401d4fb82b/src/demo/Editor.tsx#L129-L138

This won't automatically override styles for the `MenuSelectHeading` component, and it isn't super straightforward to override styles for that component. It would be nice to make that component's styles more customizable.

If you could please clarify your issue, that would help. Thanks.

**Comment by RuckieOfficial** [comment]
I agree that default fonts are huge. Now you have custom h1 font size and other font sizes are from theme. But if you use custom font sizes in project, headings > h1 could be way larger or smaller than h1.

And we use only h1 - h4 in our project with custom sizes, but you use more headings so headings > h4 could be larger or smaller too in that case.

I tried override headings in editor and it works just fine, but how u say it doesn't impact MenuSelectHeading at all which is not good, because it is not consistent then.

I hope I made it more clear now.

**Comment by mcsimps2** [comment]
Not proud of this one, but you can hack `MenuSelectHeading` like so for each of the list items.

```
            <GlobalStyles styles={(theme) => ({
                'li[data-value="Heading 1"] span': {
                    ...theme.typography.h1
                }
            })} />
```

**Comment by sjdemartini** [comment]
Heads up that as of the latest mui-tiptap version [v1.24.0](https://github.com/sjdemartini/mui-tiptap/releases/tag/v1.24.0), it should be more straightforward to override styles for the `MenuSelectHeading` component (along with essentially every component on mui-tiptap). You can do this in a few different ways. For instance:

1. You can use the following style overrides in the MUI theme and target specific slots of the component to customize however you choose. This seems most useful for your situation:

    ```ts
    const theme = createTheme({
      components: {
        "MuiTiptap-MenuSelectHeading": {
          styleOverrides: {
            headingOption1: ({theme}) => ({ fontSize: theme.typography.h1.fontSize }),
            headingOption2: ({theme}) => ({ fontSize: theme.typography.h2.fontSize }),
            headingOption3: ({theme}) => ({ fontSize: theme.typography.h3.fontSize }),
            headingOption4: ({theme}) => ({ fontSize: theme.typography.h4.fontSize }),
            headingOption5: ({theme}) => ({ fontSize: theme.typography.h5.fontSize }),
            headingOption6: ({theme}) => ({ fontSize: theme.typography.h6.fontSize }),
          },
        },
      },
    });
    ```

    In action:

    <img width="638" height="622" alt="Image" src="https://github.com/user-attachments/assets/42936efa-2513-482e-bfcd-d6c13e909769" />

2. You could alternatively target utility classes like `"MuiTiptap-MenuSelectHeading-headingOption1"` with your own custom CSS, or using `menuSelectHeadingClasses.headingOption1` if you'd like to reference the class name programmatically in JS/TS (imported like `import { menuSelectHeadingClasses } from "mui-tiptap";`).

3. You could pass in your own classes directly when rendering the component, if you have external CSS rules defined, like `<MenuSelectHeading classes={{ headingOption1: "text-2xl" }}/>`.
